src/bootstrap/src/core/build_steps/tool.rs

```diff
@@ -804,6 +805,59 @@ impl Step for LlvmBitcodeLinker {
     }
 }
 
+#[derive(Debug, Clone, Hash, PartialEq, Eq)]
+pub struct LibcxxVersionTool {
+    pub target: TargetSelection,
+}
+
+#[allow(dead_code)]
+#[derive(Debug, Clone)]
+pub enum LibcxxVersion {
+    Gnu(usize),
+    Llvm(usize),
+}
+
+impl Step for LibcxxVersionTool {
+    type Output = LibcxxVersion;
+    const DEFAULT: bool = false;
+    const ONLY_HOSTS: bool = true;
+
+    fn should_run(run: ShouldRun<'_>) -> ShouldRun<'_> {
+        run.never()
+    }
+
+    fn run(self, builder: &Builder<'_>) -> LibcxxVersion {
+        let out_dir = builder.out.join(self.target.to_string()).join("libcxx-version");
+        let _ = fs::remove_dir_all(&out_dir);
+        t!(fs::create_dir_all(&out_dir));
+
+        let compiler = builder.cxx(self.target).unwrap();
+        let mut cmd = Command::new(compiler);
+
+        let executable = out_dir.join("libcxx-version");
```

Comment:
@onur-ozkan 
```suggestion
        let executable = out_dir.join("libcxx-version.exe");
```
How did it pass CI on Windows?
It trivially fails for me locally, because there's no `libcxx-version`, only `libcxx-version.exe`.

Reply:
https://github.com/rust-lang/rust/pull/126086 should handle this.

Reply:
Maybe it passed because windows-gnu CI uses env that disables llvm download?

Reply:
> Maybe it passed because windows-gnu CI uses env that disables llvm download?

Yeah, most likely.